### PR TITLE
Add support for SE2 poses in Collision2D

### PR DESCRIPTION
**Pull request by luisenp**
There is a breaking change in this PR, since `SE2.xy()` now returns a `Point2` object instead of a tensor. 

This should be merged after #277

### CODE REVIEW

File: theseus/embodied/collision/tests/test_collision_factor.py [hunk 1]
```diff
@@ -76,13 +76,18 @@ def test_collision2d_copy():
     assert cost_function2.name == "new_name"
 
 
-def test_collision2d_jacobians():
+@pytest.mark.parametrize("pose_cls", [th.Vector, th.SE2])
```
Comment (mhmukadam) [inline theseus/embodied/collision/tests/test_collision_factor.py:79]: Why not make th.Vector -> th.Point2 directly?

File: theseus/geometry/se2.py [hunk 2]
```diff
@@ -136,7 +136,7 @@ def theta(self, jacobians: Optional[List[torch.Tensor]] = None) -> torch.Tensor:
     def translation(self) -> Point2:
         return Point2(tensor=self[:, :2])
 
-    def xy(self, jacobians: Optional[List[torch.Tensor]] = None) -> torch.Tensor:
+    def xy(self, jacobians: Optional[List[torch.Tensor]] = None) -> Point2:
```
Comment (mhmukadam) [inline theseus/geometry/se2.py:139]: I guess now it is same as translation but also gives jacobian. Maybe we add jacobian to translation and alias xy (or vice-versa)?

**Review by mhmukadam — APPROVED** [review]
Looks good.

File: theseus/embodied/collision/tests/test_collision_factor.py (hunk 1 above)
Comment (luisenp) [reply theseus/embodied/collision/tests/test_collision_factor.py:79]: Ah, sorry, when I first wrote this I wasn't sure if there was a `randn` method for `th.Point2`. I'll fix.

Comment (luisenp) [reply theseus/embodied/collision/tests/test_collision_factor.py]: Fixed.

File: theseus/geometry/se2.py (hunk 2 above)
Comment (luisenp) [reply theseus/geometry/se2.py:139]: Good catch. I'll refactor a bit.

Comment (luisenp) [reply theseus/geometry/se2.py:139]: I made `translation` an alias for `xy()`.

File: theseus/embodied/collision/tests/test_collision_factor.py (hunk 1 above)
Comment (mhmukadam) [reply theseus/embodied/collision/tests/test_collision_factor.py]: No worries. Thanks for the fix.

**Review by mhmukadam — APPROVED** [review]

**Pull request merged**